mpi/rmpi/inc/Mpi/Globals.h

```diff
@@ -0,0 +1,268 @@
+R__EXTERN const Int_t MAX_OBJECT_NAME;
+R__EXTERN const Int_t MAX_IO_BUFFER; // used for sync output in TEnvironment
+
+// environmental inquiry keys
+R__EXTERN const Int_t TAG_UB;
+R__EXTERN const Int_t HOST;
+R__EXTERN const Int_t IO;
+R__EXTERN const Int_t WTIME_IS_GLOBAL;
+R__EXTERN const Int_t APPNUM;
+R__EXTERN const Int_t LASTUSEDCODE;
+R__EXTERN const Int_t UNIVERSE_SIZE;
+R__EXTERN const Int_t WIN_BASE;
+R__EXTERN const Int_t WIN_SIZE;
+R__EXTERN const Int_t WIN_DISP_UNIT;
+
+// assorted constants
+R__EXTERN const void *BOTTOM;
+R__EXTERN const void *IN_PLACE;
+R__EXTERN const Int_t PROC_NULL;
+R__EXTERN const Int_t ANY_SOURCE;
+R__EXTERN const Int_t ROOT_RANK;
+R__EXTERN const Int_t ANY_TAG;
+R__EXTERN const Int_t UNDEFINED;
+R__EXTERN const Int_t BSEND_OVERHEAD;
+R__EXTERN const Int_t KEYVAL_INVALID;
+R__EXTERN const Int_t ORDER_C;
+R__EXTERN const Int_t ORDER_FORTRAN;
+R__EXTERN const Int_t DISTRIBUTE_BLOCK;
+R__EXTERN const Int_t DISTRIBUTE_CYCLIC;
+R__EXTERN const Int_t DISTRIBUTE_NONE;
+R__EXTERN const Int_t DISTRIBUTE_DFLT_DARG;
+
+// MPI-2 IO
+R__EXTERN const Int_t MODE_CREATE;
+R__EXTERN const Int_t MODE_RDONLY;
+R__EXTERN const Int_t MODE_WRONLY;
+R__EXTERN const Int_t MODE_RDWR;
+R__EXTERN const Int_t MODE_DELETE_ON_CLOSE;
+R__EXTERN const Int_t MODE_UNIQUE_OPEN;
+R__EXTERN const Int_t MODE_EXCL;
+R__EXTERN const Int_t MODE_APPEND;
+R__EXTERN const Int_t MODE_SEQUENTIAL;
+R__EXTERN const Int_t DISPLACEMENT_CURRENT;
+R__EXTERN const Int_t SEEK_SET;
+R__EXTERN const Int_t SEEK_CUR;
+R__EXTERN const Int_t SEEK_END;
+R__EXTERN const Int_t MAX_DATAREP_STRING;
+
+// thread constants
+R__EXTERN const Int_t THREAD_SINGLE;
+R__EXTERN const Int_t THREAD_FUNNELED;
+R__EXTERN const Int_t THREAD_SERIALIZED;
+R__EXTERN const Int_t THREAD_MULTIPLE;
+
+// results of communicator and group comparisons
+R__EXTERN const Int_t IDENT;
+R__EXTERN const Int_t CONGRUENT;
+R__EXTERN const Int_t SIMILAR;
+R__EXTERN const Int_t UNEQUAL;
+
+class TEnvironment;
+class TGroup;
+class TInfo;
+class TNullCommunicator;
+class TIntraCommunicator;
+
+// null handles
+R__EXTERN const TGroup GROUP_NULL;
+R__EXTERN const TInfo INFO_NULL;
+R__EXTERN const MPI_Datatype DATATYPE_NULL;
+R__EXTERN const TNullCommunicator COMM_NULL;
+// empty group
+R__EXTERN const TGroup GROUP_EMPTY;
+
+R__EXTERN TIntraCommunicator COMM_WORLD;
+R__EXTERN TIntraCommunicator COMM_SELF;
+
+// Functions
+
+template <class T>
+MPI_Datatype GetDataType()
+{
+   if (typeid(T) == typeid(int) || typeid(T) == typeid(Int_t)) return MPI_INT;
+   if (typeid(T) == typeid(float) || typeid(T) == typeid(Float_t)) return MPI_FLOAT;
+   if (typeid(T) == typeid(double) || typeid(T) == typeid(Double_t)) return MPI_DOUBLE;
+   if (typeid(T) == typeid(bool) || typeid(T) == typeid(Bool_t)) return MPI_BYTE;
+   if (typeid(T) == typeid(char) || typeid(T) == typeid(Char_t)) return MPI_CHAR;
+   if (typeid(T) == typeid(short) || typeid(T) == typeid(Short_t)) return MPI_SHORT;
```

Comment:
Hi, couldn't you just specialize that template for the different types? I don't think using `typeid` on a template type is necessary.